scene/gui/tab_container.cpp

```diff
@@ -417,7 +419,13 @@ void TabContainer::set_current_tab(int p_current) {
 	}
 
 	_change_notify("current_tab");
-	emit_signal("tab_changed", current);
+	if (pending_previous == current)
+		emit_signal("tab_selected", current);
```

Comment:
The output for the new code is:
**Code**
```
	emit_signal("tab_selected", current);
	if (pending_previous != current) {
		previous = pending_previous;
		emit_signal("tab_changed", current);
	}
```
**Output** 
```
on_tab_selected: current tab index 1, previous tab index: 0
on_tab_changed: current tab index 1, previous tab index: 0

on_tab_selected: current tab index 0, previous tab index: 0
on_tab_changed: current tab index 0, previous tab index: 1

on_tab_selected: current tab index 1, previous tab index: 1
on_tab_changed: current tab index 1, previous tab index: 0

on_tab_selected: current tab index 2, previous tab index: 0
on_tab_changed: current tab index 2, previous tab index: 1

on_tab_selected: current tab index 3, previous tab index: 1
on_tab_changed: current tab index 3, previous tab index: 2

on_tab_selected: current tab index 4, previous tab index: 2
on_tab_changed: current tab index 4, previous tab index: 3
```

It should be like this:
**Code**
```
	if (pending_previous == current)
		emit_signal("tab_selected", current);
	else {
		previous = pending_previous;
		emit_signal("tab_selected", current);
		emit_signal("tab_changed", current);
	}
```

**Output**
```
on_tab_selected: current tab index 1, previous tab index: 0
on_tab_changed: current tab index 1, previous tab index: 0

on_tab_selected: current tab index 0, previous tab index: 1
on_tab_changed: current tab index 0, previous tab index: 1

on_tab_selected: current tab index 1, previous tab index: 0
on_tab_changed: current tab index 1, previous tab index: 0

on_tab_selected: current tab index 2, previous tab index: 1
on_tab_changed: current tab index 2, previous tab index: 1

on_tab_selected: current tab index 3, previous tab index: 2
on_tab_changed: current tab index 3, previous tab index: 2

on_tab_selected: current tab index 4, previous tab index: 3
on_tab_changed: current tab index 4, previous tab index: 3
```

Did you notice the difference? `previous` tab index is not updated. That is what drove me crazy, it seems the putting the emission outside the if statement is a benign thing. But there is a very subtle difference. Look carefully when `previous` gets updated by `pending_previous`. Not necessarily every time `set_current_tab()` is called. 

But I am not crazy, lol, I've been programming since the days of Atari ST, MSX and Commodore 64, even assembly of Z80 processor when I was 9. This is the first time a silly fragment of code gave such headache.

I am reverting to previous code.